cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,87 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
-describe('Settings page', () => {
-  before(() => {
+import { beforeEach } from "mocha";
+import PageObject from "../support/PageObject";
+import HomePageObject from "../support/pages/home.pageObject";
+import SettingsPageObject from "../support/pages/settings.pageObject";
+import SignInPageObject from "../support/pages/signIn.pageObject";
 
-  });
+const homePage = new HomePageObject;
+const settingsPage = new SettingsPageObject;
+const signInPage = new SignInPageObject;
+
+
+describe('Settings page', () => {
+  let user;
 
   beforeEach(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
 
   });
 
+  beforeEach(() => {
+    cy.login(user.email, user.username, user.password);
+    settingsPage.visit();
+  })
+
   it('should provide an ability to update username', () => {
+   
+    settingsPage.usernameField.clear().type(user.updateUsername);
```

Comment:
move this method to the page object, like this
```suggestion
    settingsPage.fillUsernameField(user.updateUsername);
```